apis/python/src/tiledb/vector_search/vamana_index.py

```diff
@@ -0,0 +1,154 @@
+        resources: Optional[Mapping[str, Any]] = None
+    ):
+        """
+        Query an VAMANA index
+
+        Parameters
+        ----------
+        queries: numpy.ndarray
+            ND Array of queries
+        k: int
+            Number of top results to return per query
+        nthreads: int
+            Number of threads to use for query
+        mode: Mode
+            If provided the query will be executed using TileDB cloud taskgraphs.
+            For distributed execution you can use REALTIME or BATCH mode. 
+            For local execution you can use LOCAL mode.
+        resource_class: 
+            The name of the resource class to use ("standard" or "large"). Resource classes define maximum
+            limits for cpu and memory usage. Can only be used in REALTIME or BATCH mode.
+            Cannot be used alongside resources.
+            In REALTIME or BATCH mode if neither resource_class nor resources are provided,
+            we default to the "large" resource class.
+        resources:
+            A specification for the amount of resources to use when executing using TileDB cloud 
+            taskgraphs, of the form: {"cpu": "6", "memory": "12Gi", "gpu": 1}. Can only be used 
+            in BATCH mode. Cannot be used alongside resource_class.
+        """
+        if self.size == 0:
+            return np.full((queries.shape[0], k), index.MAX_FLOAT_32), np.full(
+                (queries.shape[0], k), index.MAX_UINT64
+            )
+
+        if mode != Mode.BATCH and resources:
+            raise TypeError("Can only pass resources in BATCH mode")
+        if (mode != Mode.REALTIME and mode != Mode.BATCH) and resource_class:
+            raise TypeError("Can only pass resource_class in REALTIME or BATCH mode")
+
+        assert queries.dtype == np.float32
+
+        if queries.ndim == 1:
+            queries = np.array([queries])
+
+        if nthreads == -1:
+            nthreads = multiprocessing.cpu_count()
+
+        # TODO(paris): Actually run the query.
+        return [], []
+
+# TODO(paris): Pass more arguments to C++, i.e. storage_version.
+def create(
+    uri: str,
+    dimensions: int,
+    vector_type: np.dtype,
+    id_type: np.dtype = np.uint32,
+    adjacency_row_index_type: np.dtype = np.uint32,
+    group_exists: bool = False,
+    config: Optional[Mapping[str, Any]] = None,
+    storage_version: str = STORAGE_VERSION,
+    **kwargs,
+) -> VamanaIndex:
+      if not group_exists:
+        ctx = vspy.Ctx(config)
+        index = vspy.IndexVamana(
+            feature_type=np.dtype(vector_type).name, 
+            id_type=np.dtype(id_type).name, 
+            adjacency_row_index_type=np.dtype(adjacency_row_index_type).name, 
+            dimension=dimensions,
+        )
+        empty_vector = vspy.FeatureVectorArray(
+            dimensions, 
+            0, 
+            np.dtype(vector_type).name, 
+            np.dtype(id_type).name
+            )
+        index.train(empty_vector)
```

Comment:
I think we should try to simplify this code by providing one method in the C++ API that creates an empty index.

Reply:
Agree, though for now I'd prefer to leave as-is just for now - I don't yet have everything working end to end and it's faster to make changes in Python than in native code. Added a TODO and will update once it's fully working, if okay with you?